osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/mbus/CoupleMbusDeviceCommandExecutorTest.java

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright 2021 Alliander N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.mbus;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.DlmsDevice;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.Protocol;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.factories.DlmsConnectionManager;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ChannelElementValuesDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.MbusChannelElementsDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.MbusChannelElementsResponseDto;
+
+@ExtendWith(MockitoExtension.class)
+public class CoupleMbusDeviceCommandExecutorTest {
+
+    @Mock
+    private DeviceChannelsHelper deviceChannelsHelper;
+
+    @Mock
+    private DlmsConnectionManager conn;
+
+    @Mock
+    private DlmsDevice device;
+
+    @InjectMocks
+    private CoupleMBusDeviceCommandExecutor commandExecutor = new CoupleMBusDeviceCommandExecutor();
+
+    private short channel = (short) 1;
+    private Short primaryAddress = 9;
+    private String mbusDeviceIdentification = "G00009";
+    private String manufacturerIdentification = "manufacturerIdentification";
+    private short version = 123;
+    private short deviceTypeIdentification = 456;
+    private String identificationNumber = "identificationNumber";
+    private List<ChannelElementValuesDto> candidateChannelElementValues = Arrays
+            .asList(new ChannelElementValuesDto(this.channel, this.primaryAddress, this.identificationNumber,
+                    this.manufacturerIdentification, this.version, this.deviceTypeIdentification));
+
+    @Test
+    public void testExactMatch() throws ProtocolAdapterException {
```

Comment:
these tests look very similar to the "ByChannelCommandExecutorTest" . is there a way to make super class here?

Reply:
it looks quite similar but it's not. Testing data objects of different classes